File: src/main/java/fr/adrienbrault/idea/symfony2plugin/action/NewCommandAction.java
package fr.adrienbrault.idea.symfony2plugin.action;

import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiDirectory;
import fr.adrienbrault.idea.symfony2plugin.Symfony2ProjectComponent;
import org.jetbrains.annotations.NotNull;

import java.util.Map;

import static fr.adrienbrault.idea.symfony2plugin.util.StringUtils.underscore;

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 */
public class NewCommandAction extends AbstractNewPhpClassAction {
    public NewCommandAction() {
        super("Command", "Create Command Class");
    }

    @Override
    protected String getClassNameSuffix() {
        return "Command";
    }

    @Override
    protected String getTemplateName(@NotNull Project project, @NotNull String namespace, @NotNull PsiDirectory directory) {
        return NewFileActionUtil.guessCommandTemplateType(project, namespace);
    }

    @Override
    protected Map<String, String> getExtraVariables(@NotNull String className, @NotNull String namespace, @NotNull PsiDirectory directory) {
        String clazz = className.endsWith("Command")
            ? className.substring(0, className.length() - "Command".length())
            : className;
        String prefix = NewFileActionUtil.getCommandPrefix(directory);
        return Map.of("command_name", prefix + ":" + underscore(clazz));
    }

    public static class Shortcut extends NewCommandAction {
        @Override
        public void update(@NotNull AnActionEvent event) {
            this.setStatus(event, false);
            Project project = getEventProject(event);
            if (!Symfony2ProjectComponent.isEnabled(project)) {
                return;
            }

            this.setStatus(event, NewFileActionUtil.isInGivenDirectoryScope(event, "Command"));
        }
    }
}

File: src/main/kotlin/fr/adrienbrault/idea/symfony2plugin/action/NewCommandAction.kt
package fr.adrienbrault.idea.symfony2plugin.action

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.project.Project
import com.intellij.psi.PsiDirectory
import fr.adrienbrault.idea.symfony2plugin.Symfony2ProjectComponent
import fr.adrienbrault.idea.symfony2plugin.util.StringUtils.underscore

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 */
open class NewCommandAction : AbstractNewPhpClassAction("Command", "Create Command Class") {
    override fun getClassNameSuffix() = "Command"

    override fun getTemplateName(project: Project, namespace: String, directory: PsiDirectory): String =
        NewFileActionUtil.guessCommandTemplateType(project, namespace)

    override fun getExtraVariables(
        className: String,
        namespace: String,
        directory: PsiDirectory
    ): Map<String, String> {
        val clazz = className.removeSuffix("Command")
        val prefix = NewFileActionUtil.getCommandPrefix(directory)
        return mapOf("command_name" to "$prefix:${underscore(clazz)}")
    }

    class Shortcut : NewCommandAction() {
        override fun update(event: AnActionEvent) {
            setStatus(
                event,
                Symfony2ProjectComponent.isEnabled(getEventProject(event)) &&
                        NewFileActionUtil.isInGivenDirectoryScope(event, "Command")
            )
        }
    }
}
